pkg/kubectl/cmd/annotate.go

```diff
@@ -148,7 +148,7 @@ func NewCmdAnnotate(parent string, f cmdutil.Factory, ioStreams genericclioption
 	cmd.Flags().StringVar(&o.resourceVersion, "resource-version", o.resourceVersion, i18n.T("If non-empty, the annotation update will only succeed if this is the current resource-version for the object. Only valid when specifying a single resource."))
 	usage := "identifying the resource to update the annotation"
 	cmdutil.AddFilenameOptionFlags(cmd, &o.FilenameOptions, usage)
-	cmdutil.AddDryRunFlag(cmd)
+	cmdutil.SetupDryRun(cmd)
```

Comment:
This seems like a perfect candidate to live in [cli-runtime](https://github.com/kubernetes/kubernetes/blob/52994701526a779d6a19faec66f0ca8fb064c7c6/staging/src/k8s.io/cli-runtime/). See [RecordFlags](https://github.com/kubernetes/kubernetes/blob/52994701526a779d6a19faec66f0ca8fb064c7c6/staging/src/k8s.io/cli-runtime/pkg/genericclioptions/record_flags.go#L19) and how they are used in [run.go](https://github.com/kubernetes/kubernetes/blob/52994701526a779d6a19faec66f0ca8fb064c7c6/pkg/kubectl/cmd/run.go#L105). It would be nice to follow the patterns set there as well, in which case the method should be `AddFlags`.

Reply:
I do hope at some point in time to entirely (or almost entirely) get rid of `cmdutil`, so that it holds only kubectl specific code.